dp/generate_climos.py

```diff
@@ -118,72 +130,50 @@ def climo_outputs(time_resolution):
                              .format(ops_by_resolution.keys(), time_resolution))
 
     logger.info('Forming climatological means')
-    climo_means = cdo.copy(input=' '.join(climo_outputs(input_file.time_resolution)))
-
-    # Post-process climatological means
-    if convert_longitudes:
-        convert_longitude_range(climo_means)
+    climo_means_files = climo_outputs(input_file.time_resolution)
 
-    climo_means = convert_pr_var_units(input_file, climo_means)
+    # Optionally concatenate means for each interval (month, season, year) into one file
+    if not split_intervals:
+        logger.info('Concatenating mean interval files')
+        climo_means_files = [cdo.copy(input=' '.join(climo_means_files))]
 
-    # Update climo file with climo specific metadata attributes.
-    # Do it in place via CFDataset to avoid CDO installation hassles: CDO < 1.8.0 does not have setattributes method
-    # and there's no easy Ubuntu install for CDO >= 1.8.0 yet
-    # Also avoids copying the file just to update its attributes.
-    with CFDataset(climo_means, mode='r+') as cf:
-        # For an explanation of frequency param, see PCIC metadata standard
-        resolution_to_frequency = {
-            'daily': 'msaClim',
-            'monthly': 'saClim',
-            'yearly': 'aClim'
-        }
-        try:
-            cf.frequency = resolution_to_frequency[input_file.time_resolution]
-        except KeyError:
-            raise ValueError("Expected input file to have time resolution in {}, found '{}'"
-                             .format(resolution_to_frequency.keys(), input_file.time_resolution))
-        # In Python2.7, datetime.datime.isoformat does not take params telling it how much precision to
-        # provide in its output; standard requires 'seconds' precision, which means the first 19 characters.
-        cf.climo_start_time = t_start.isoformat()[:19] + 'Z'
-        cf.climo_end_time = t_end.isoformat()[:19] + 'Z'
-        if hasattr(input_file, 'tracking_id'):
-            cf.climo_tracking_id = input_file.tracking_id
-
-    # Update time metadata in climo file
-    update_climo_time_meta(climo_means)
-
-    # Create final output file(s): split climo means file by dependent variables if required
+    # Optionally convert longitudes in each file
+    if convert_longitudes:
+        logger.info('Converting longitudes')
+        for climo_means_file in climo_means_files:
+            convert_longitude_range(climo_means_file)
```

Comment:
Better if `convert_longitude_range` returned the file name, and we use a list comprehension to invoke it, so that if its implementation changes (e.g., to use CDO) this code will be robust.